test/common/upstream/upstream_impl_test.cc

```diff
@@ -1566,7 +1566,7 @@ TEST_F(StaticClusterImplTest, UnsupportedLBType) {
     connect_timeout: 0.25s
     type: static
     lb_policy: fakelbtype
-    hosts:
+    hidden_envoy_deprecated_hosts:
```

Comment:
I don't think we should be testing this field like this? Either convert or use the deprecated field with a deprecated field test? (We should make sure we still have tests that cover the conversion path)

Reply:
I fixed this one. There are actually a bunch of places where `hosts:` still appears in this file. These are never going to be deprecated tests as I don't think we plan to turn down v2 hosts until v2 disappears. So, I think we have sufficient coverage. The reason this specific test was problematic was due to this being upgraded to v3 due to an invalid LB type and then failing the config check for the wrong reason at v3.

Reply:
I'm confused, doesn't marking it deprecated cause the fail on warn tests to fail? Or is the issue that those tests don't run through the validation path? I would expect to see a DEPRECATED_FEATURE_TEST here unless the config is not going through the validation path?

Reply:
It would fail due to this, except it fails even earlier, because `lb_policy` has an invalid value (which is what the test is trying to catch).

Reply:
As a follow up shouldn't we fix all these tests even though they fail for some other reason? Mainly just for reduced confusion in the future?

Reply:
For `hosts` specifically, I think we are supporting this forever for v2, we're not going to disable-by-default, so it's fine to keep using this. We probably do need a bucket of work in moving the trailing example configs/docs/unit tests that still use v2 to v3, but I think that's out of scope for this PR.

Reply:
I think we are talking past each other here, even without disable-by-default, we still have DEPRECATED_FEATURE_TEST and a compile time options build that fails on any use of a deprecated feature, whether disabled by default or not. My point is that I would expect any use of `host` to fail in this build, but I think it's failing for other reasons. We should still clean up hosts in these tests otherwise it's debt that will be confusing later. cc @alyssawilk for further comment.

Reply:
Yeah, basically I think this config would be fatal anyway, but as Matt points out, we hit the fatal condition we care about first so it's not a problem as far as gunit is concerned.
I agree with Matt that I'd prefer we clean configs up as much as possible (if for no other reason than for folks who copy-paste-edit) but I don't think it needs to be release blocking

I think in general once v3 lands, we need to do a sweep for v2 config and stop using it apart from coverage checks.  We have to do the clean up eventually and doing it sooner rather than later avoids hacks like this.  I think that'd resolve this concern since hosts isn't in v3 at all.